api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -227,23 +235,16 @@ public void updateProfileForUser(DbUser user, Profile updatedProfile, Profile pr
     profileAuditor.fireUpdateAction(previousProfile, appliedUpdatedProfile);
   }
 
-  public void validateAndCleanProfile(Profile profile) throws BadRequestException {
-    // Validation steps, which yield a BadRequestException if errors are found.
-    String userName = profile.getUsername();
-    if (userName == null || userName.length() < 3 || userName.length() > 64) {
-      throw new BadRequestException(
-          "Username should be at least 3 characters and not more than 64 characters");
-    }
-    validateStringLength(profile.getGivenName(), "Given Name", 80, 1);
-    validateStringLength(profile.getFamilyName(), "Family Name", 80, 1);
-
+  public void cleanProfile(Profile profile) throws BadRequestException {
```

Comment:
I like the name change. It makes expect changes.

I would prefer a more immutable style, though, where we take in a profile and return a new one, especially since these aren't ORM objects.

Reply:
We can  make a `cloneProfile()` easily with MapStruct. You just have a method that takes and returns a Profile.